src/main/java/com/google/edith/servlets/UserStatsServlet.java

```diff
@@ -33,7 +33,7 @@
 
   public UserStatsServlet() {
     this.datastore = DatastoreServiceFactory.getDatastoreService();
-    this.userInsights = new UserInsights("userId");
+    this.userInsights = new UserInsightsService("userId");
```

Comment:
Since we have a service for this servlet. Can we move other external dependencies like datastore to the service ? And the service will expose all the methods necessary interacting with datastore.
This will also help in testing. And you can mock the service and just tests logic only for the servlet.

Reply:
Yea!